controllers/machineset.go

```diff
@@ -98,187 +79,531 @@ func scaleMachineSet(aw *arbv1.AppWrapper, userRequestedInstanceType string, rep
+	// Retrieve the machine object
+	machine := &machinev1beta1.Machine{}
+	err := r.Get(ctx, types.NamespacedName{Name: machineName, Namespace: namespaceToList}, machine)
+	if err != nil {
+		klog.Infof("Error retrieving machine: %v", err)
+		return err
+	}
+
+	// Apply the patch to remove the label
+	patch := client.RawPatch(types.JSONPatchType, patchBytes)
+	err = r.Patch(ctx, machine, patch)
 	if err != nil {
-		klog.Infof("deleted label to machines %v", err)
+		klog.Infof("Error removing label from machine: %v", err)
+		return err
 	}
+	return nil
 }
 
-func removeLabelFromNode(aw *arbv1.AppWrapper, nodeName string) {
+func (r *AppWrapperReconciler) removeLabelFromNode(ctx context.Context, aw *arbv1.AppWrapper, nodeName string) error {
 	labelPatch := fmt.Sprintf(`[{"op":"remove","path":"/metadata/labels/%s","value":"%s" }]`, aw.Name, aw.Name)
-	_, err := kubeClient.CoreV1().Nodes().Patch(context.Background(), nodeName, types.JSONPatchType, []byte(labelPatch), metav1.PatchOptions{})
+	_, err := r.kubeClient.CoreV1().Nodes().Patch(ctx, nodeName, types.JSONPatchType, []byte(labelPatch), metav1.PatchOptions{})
 	if err != nil {
 		klog.Infof("Error deleted label to machines %v", err)
+		return err
 	}
+	return nil
 }
 
 // add logic to swap out labels with new appwrapper label
-func swapNodeLabels(oldAw *arbv1.AppWrapper, newAw *arbv1.AppWrapper) {
-	allMachines, errm := machineClient.MachineV1beta1().Machines(namespaceToList).List(context.Background(), metav1.ListOptions{})
+func (r *AppWrapperReconciler) swapNodeLabels(ctx context.Context, oldAw *arbv1.AppWrapper, newAw *arbv1.AppWrapper) error {
+	labelSelector := labels.SelectorFromSet(labels.Set(map[string]string{
+		oldAw.Name: oldAw.Name,
+	}))
+	listOptions := &client.ListOptions{
+		LabelSelector: labelSelector,
+	}
+	allMachines := machinev1beta1.MachineList{}
+	errm := r.List(ctx, &allMachines, listOptions)
 	if errm != nil {
 		klog.Infof("Error creating machineset: %v", errm)
+		return errm
 	}
 	//klog.Infof("Got all machines %v", allMachines)
-	for idx := range allMachines.Items {
-		machine := &allMachines.Items[idx]
-		for k, _ := range machine.Labels {
-			if k == oldAw.Name {
-				nodeName := machine.Status.NodeRef.Name
-				klog.Infof("removing label from node %v that belonged to appwrapper %v and adding node to new appwrapper %v", nodeName, oldAw.Name, newAw.Name)
-				removeLabelFromMachine(oldAw, machine.Name)
-				removeLabelFromNode(oldAw, nodeName)
-				addLabelToMachine(newAw, machine.Name)
-				addLabelToNode(newAw, nodeName)
-			}
-		}
+	for _, machine := range allMachines.Items {
+		nodeName := machine.Status.NodeRef.Name
+		klog.Infof("removing label from node %v that belonged to appwrapper %v and adding node to new appwrapper %v", nodeName, oldAw.Name, newAw.Name)
+		r.removeLabelFromMachine(ctx, oldAw, machine.Name)
+		r.removeLabelFromNode(ctx, oldAw, nodeName)
+		r.addLabelToMachine(ctx, newAw, machine.Name)
+		r.addLabelToNode(ctx, newAw, nodeName)
```

Comment:
The `removeLabelFromNode` and `addLabelToNode` patches can be done in a single atomic patch. So these two methods can be a single one.

Also the returned error should be handled.